core/src/impl/Kokkos_Core.cpp

```diff
@@ -165,6 +165,26 @@ bool is_valid_map_device_id_by(std::string const& x) {
 
 }  // namespace
 
+[[nodiscard]] int Kokkos::device_id() noexcept {
+#if defined(KOKKOS_ENABLE_CUDA)
+  return Cuda().cuda_device();
+#elif defined(KOKKOS_ENABLE_HIP)
+  return HIP().hip_device();
+#elif defined(KOKKOS_ENABLE_OPENACC)
+  return Experimental::OpenACC().acc_device_number();
+#elif defined(KOKKOS_ENABLE_OPENMPTARGET)
```

Comment:
I'm guessing OPENMPTARGET is like a device so both Cuda and OpenMP cannot be returned at the same time. No need to check KOKKOS_ENABLE_OPENMP here?

Reply:
OpenMP is a "host" backend so it has no influence on the "device id"